pkg/vpcmodel/explainabilityConnectivity.go

```diff
@@ -329,7 +327,8 @@ func mergeAllowDeny(allow, deny rulesInLayers) rulesInLayers {
 			default: // no rules
 				rType = NoRules
 			}
-			mergedRulesInFilter := RulesInTable{TableIndex: filterIndex, Rules: mergedRules, RulesOfType: rType}
+			mergedRulesInFilter := RulesInTable{TableIndex: filterIndex, Rules: mergedRules, RulesOfType: rType,
+				TableHasEffect: allowRules.TableHasEffect} // TableHasEffect can be taken from either allow or deny
```

Comment:
TableHasEffect is table dependent. 
1. I did document it in `abstractVPC`, see below
2. Not perfect, I agree. I don't see a better solution; we can discuss and see if you have a better idea 

```
// RulesInTable for a given layer (SGLayer/NACLLayer) or transit gateway contains
// index of the SG/NACL/transit gateway filter and the indexes of the rules within it
// this struct is also used for intermediate computation of only allow or only deny rules; TableHasEffect,
// TableConn are always w.r.t. the entire table
```

Reply:
Good catch. There is an issue here, but not the one you mention.
We get to the relevant section of the code - merge - only if the layer has both allow and deny rules.  That could be only NACL. And since a single NACL is associated with a subnet, if we got here there is a single table in the layer, which has both allow and deny rules. Hence the assumption is correct. However, the code is too complicated, confusing and even buggy: had we had 2 NACL one with both deny and allow and one with allow only, the code would dump. This will never happen, but clearly the code should be fixed (and simplified(. 
I added a todo and opened an issue I will handle it I am done with https://github.com/np-guard/vpc-network-config-analyzer/issues/859